sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UriShouldNotBeHardcoded.cs

```diff
@@ -0,0 +1,146 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class UriShouldNotBeHardcoded : SonarDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S1075";
+        private const string MessageFormat = "{0}";
+        private const string AbsoluteUriMessage = "Refactor your code to get this URI from a customizable parameter.";
```

Comment:
I don't like this message. "customizable parameter" sounds weird. Could parameter be not customizable?
Perhaps: "Refactor your code not to use hardcoded path or URI."?

Also, I prefer "hardcoded" over "hard-coded" everywhere.